pandera/dtypes.py

```diff
@@ -225,13 +233,17 @@ def from_python_type(cls, python_type: type) -> "PandasDtype":
             str: cls.String,
             int: cls.Int,
             float: cls.Float,
+            complex: cls.Complex128  # np.complex is alias for np.complex128
         }.get(python_type)
 
         if pandas_dtype is None:
-            raise TypeError(
-                "python type '%s' not recognized as pandas data type" %
-                python_type
-            )
+            if python_type in [item for sublist in np.sctypes.values() for item in sublist]:
```

Comment:
for clarity, I think this functionality should be in a separate `from_numpy_type` method, see #262, this method should be reserved for built-in python types.

To handle numpy scalar types, see here: https://github.com/pandera-dev/pandera/blob/master/pandera/schemas.py#L775-L793

Would just need to call `PandasDtype.from_numpy_types` in the conditional elif statements when the supplied value is in `np.sctypes.values()`. Those would also benefit from being defined as a global var in the `dtypes` module as a set:

```python
NUMPY_TYPES = frozenset(np.sctypes.values())


class PandasDtype:
    ...
    def from_numpy_type(cls, numpy_type: type):
        ...


# then in SeriesSchemaBase.dtype:

def dtype():
    ...
    elif self._pandas_dtype in NUMPY_TYPES:
        dtype = PandasDtype.from_numpy_type(self._pandas_dtype).str_alias
    ...
```

I think we can close out both #261 #262 in this PR

Reply:
Yes, I agree. I made the changes.